ipn/ipnauth/actor.go

```diff
@@ -45,3 +48,51 @@ type ActorCloser interface {
 	// Close releases resources associated with the receiver.
 	Close() error
 }
+
+// SessionID is an opaque, comparable value used to identify a logical [Session].
+type SessionID struct {
+	v any
+}
+
+type uniqueID struct {
+	p *byte
+	s string
+}
+
+// NewSessionID returns a new process-unique [SessionID].
+func NewSessionID() SessionID {
+	// We use both a byte pointer and a random hex string out of paranoia that
+	// eventually we may use a SessionID in a context where 64 bits of entropy
+	// of the hex string is not enough. The string is nice for debugging and logging,
+	// but the pointer guarantees uniqueness of the ID within the address space
+	// of a given process.
+	return SessionID{uniqueID{new(byte), rands.HexString(16)}}
+}
+
+// SessionIDFrom returns a new [SessionID] derived from the specified value.
+// SessionIDs derived from equal values are equal.
+func SessionIDFrom[T comparable](v T) SessionID {
+	return SessionID{v}
+}
+
+// String implements [fmt.Stringer].
+func (id SessionID) String() string {
+	if id.v == nil {
+		return "(none)"
+	}
+	return fmt.Sprint(id.v)
+}
+
+func (id uniqueID) String() string {
+	return id.s
+}
+
+// Session is an optional interface that an [Actor] might implement to link a
+// series of related interactions with the LocalAPI by a [SessionID].
+// It doesn't necessarily correspond to the same [net.Conn] or any physical session.
+// For example, all interactions with the same client process, if one is known,
+// could be considered part of the same session.
+type Session interface {
```

Comment:
Per chat, let's move this to a method on Actor that returns (_ Foo, ok bool) where Foo is some new name for "Session" that's less ambiguous with our existing session concept.

Maybe `PIDHandle`? It's a unique/opaque handle to a pid (the tuple `(pid, pidStartTime)` internally on Windows or FreeBSD, or maybe a `pidfd` on Linux/etc later, or the sameuserproof token on macOS)

I don't love `PIDHandle`, though, but it's not "Session" at least.

Still thinking about a name.

Reply:
Done.

However, I’m not a fan of `PIDHandle` either, for two main reasons:
1. The `Handle` part could be misleading, as it might imply it's related to Windows handles.
2. It won’t always represent a PID or specific process in general. You already mentioned the sameuserproof on macOS, and at some point, we may need to differentiate between different users/sessions utilizing the LocalAPI via the web client (including [remotely over Tailscale](https://tailscale.com/kb/1325/device-web-interface#open-and-authenticate-to-the-web-interface)).

How about `ClientID` as the name?